### PR TITLE
Add package coq-file-sync

### PR DESCRIPTION
**Pull request by liyishuai**

### CODE REVIEW

File: extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+maintainer: "Yishuai Li <yishuai@cis.upenn.edu>"
+version: "dev"
```
Comment (palmskog) [inline extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam:3]:
```suggestion
```

File: extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+tags: [
+  "logpath:FileSync"
+]
+authors: [
+  "Yishuai Li <yishuai@cis.upenn.edu>"
+]
+
+url {
+  src: "git+https://github.com/liyishuai/file-sync"
```
Comment (palmskog) [inline extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam:34]:
```suggestion
  src: "git+https://github.com/liyishuai/file-sync.git"
```

**Review by palmskog — COMMENTED** [review]
@liyishuai the package doesn't seem to build due to some problem with a dependency. Could you check it out so we can get a green build, e.g., on OCaml 4.13.0?

**Comment by liyishuai** [comment]
Now I'm totally confused. How did `conf-gtk2` come out?

**Comment by palmskog** [comment]
@liyishuai recall that the opam archive's CI runs tests, and thus installs all test dependencies. It looks like the `unison` package requires `lablgtk2`, which in turn requires `conf-gtk2`.

**Comment by palmskog** [comment]
@liyishuai since the only failure is seemingly `unison`, and this is dev package, I can still merge this package if you want. However, are you sure it's a good idea to have tests that require a graphics environment in the long run?

**Comment by silene** [comment]
I wonder if the following line should not be changed to include `--with-test`, because the current output is quite confusing.

https://github.com/coq/opam-coq-archive/blob/79449c56700110a9941f04e8be197ce36498c584/scripts/opam-coq-install-remove#L60

**Comment by liyishuai** [comment]
I have `lablgtk2` working on my Ubuntu with OCaml 4.13.1. Does the CI machine lack `gtk+-2.0`? If so, no package that depend on `unison` would work.
Two solutions:
1. Install `gtk+-2.0` on CI
2. Make a separate Unison package that only includes command line interface.

**Comment by palmskog** [comment]
We *can* install `gtk2+-2.0` in CI, but I'd rather not, since it's a big dependency that I don't think we want to have around. The ideal would be a CLI-only `unison`, but that may take a while. As I said we can still merge the package with the test failures, if that's the best solution for now...

File: extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+build: [make "-j%{jobs}%"]
+run-test: [make "-j%{jobs}%" "test"]
+install: [make "install" "INSTALLDIR=%{bin}%"]
+depends: [
+  "ocaml" { >= "4.12" }
+  "coq" { >= "8.14~" < "8.16" }
+  "coq-async-test" 
+  "ocamlbuild" 
+  "fileutils" 
+  "unison" { with-test }
```
Comment (liyishuai) [inline extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam:22]: or remove this dependency for now?
```suggestion
```

File: extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+dev-repo: "git+https://github.com/liyishuai/file-sync.git"
+bug-reports: "https://github.com/liyishuai/file-sync/issues"
+license: "MPL-2.0"
+
+synopsis: "File synchroniser specification in Coq"
+description: """
+Coq formalisation of the Unison file synchroniser"""
+
+build: [make "-j%{jobs}%"]
+run-test: [make "-j%{jobs}%" "test"]
```
Comment (palmskog) [inline extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam:14]:
```suggestion
```
If you remove the test dependency, you will probably have failing test unless we also do this.

Comment (liyishuai) [inline extra-dev/packages/coq-file-sync/coq-file-sync.dev/opam:14]: I see. `run-test` should be skipped as well.
```suggestion
```

**Pull request merged**